src/java/za/co/mip/ablduck/ABLDuckClassVisitor.java

```diff
@@ -0,0 +1,214 @@
+                js.meta.deprecated.version = commentParseResult.deprecatedVersion;
+                js.meta.deprecated.text    = commentParseResult.deprecatedText;
+            }
+
+        } catch (IOException ex) {
+            throw ex;
+        }
+
+        Integer methodCount = 0;
+        String previousMethodName = "";
+        // Members
+        for (Method method : cu.methods) {
+            MemberObject m = new MemberObject();
+
+            if (previousMethodName.equals(method.methodName)) {
+                methodCount++;
+            } else {
+                methodCount = 0;
+                previousMethodName = method.methodName;
+            }
+
+            if (methodCount == 0)
+                m.id = "method-" + method.methodName;
+            else
+                m.id = "method-" + method.methodName + "-" + methodCount.toString();
+            m.name = method.methodName;
+            m.owner = fullClassName;
+            m.tagname = "method";
+            m.signature = method.signature;
+
+            try {
+                CommentParseResult commentParseResult = CommentParser.parseComment(method.methodComment, fullClassName + ":" + method.methodName);
+
+                if (!commentParseResult.internal)
+                    commentParseResult.internal = method.methodName.startsWith("_");
+
+                m.returnComment = commentParseResult.returnComment;
+                m.comment = commentParseResult.comment;
+
+                if (commentParseResult.internal)
+                    m.meta.internal = "This is a private method for internal use by the framework. Don't rely on its existence.";
+
+                //TODO: remove this duplication tsk tsk, write a meta adding method for class, method and property
+                if (!commentParseResult.deprecatedVersion.equals("")) {
+                    m.meta.deprecated = new DeprecatedObject();
+        
+                    m.meta.deprecated.version = commentParseResult.deprecatedVersion;
+                    m.meta.deprecated.text    = commentParseResult.deprecatedText;
+                }
+            } catch (IOException ex) {
+                throw ex;
+            }
+            
+            switch (method.modifier){
+                case PRIVATE:
+                    m.meta.private_ = true;
+                    break;
+                case PROTECTED:
+                    m.meta.protected_ = true;
+                    break;
+            }
+            
+            if (method.isAbstract)
+                m.meta.abstract_ = true;
+
+            if (method.isStatic)
+                m.meta.static_ = true;
+
+            m.parameters = new ArrayList<ParameterObject>();
+            for (Parameter parameter : method.parameters) {
+                ParameterObject p = new ParameterObject();
+
+                p.name = parameter.name;
+                p.datatype = parameter.dataType;
+
+                //TODO: do i need this? probably need to show this in the docs somewhere
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1135)